From 1ac7b1d4ccf4409b2a52f26feaced69e5063c6ae Mon Sep 17 00:00:00 2001
From: Han Gao <gaohan@iscas.ac.cn>
Date: Sun, 17 May 2026 04:04:59 +0800
Subject: [PATCH 101/102] RUYI: riscv: dts: pine64-starpro64: add 1.6 GHz CPU
 OPP

Extend the shared CPU OPP table with a 1.6 GHz entry at 0.9 V on the
Pine64 StarPro64 board. The base eic7700 OPP table caps the CPU at
1.4 GHz; Pine64 StarPro64 is validated to run at 1.6 GHz.

Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
---
 arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts b/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts
index 7f7daf785fac..27107a5b6a8e 100644
--- a/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts
+++ b/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts
@@ -49,6 +49,14 @@ vcc_1v8: vcc1v8 {
 	};
 };
 
+&cpu_opp_table {
+	opp-1600000000 {
+		opp-hz = /bits/ 64 <1600000000>;
+		opp-microvolt = <900000>;
+		clock-latency-ns = <70000>;
+	};
+};
+
 &xtal24m {
 	clock-frequency = <24000000>;
 	clock-output-names = "xtal24m";
-- 
2.47.3

